extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/MethodNameParser.java

```diff
@@ -0,0 +1,475 @@
+package io.quarkus.spring.data.deployment;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.ClassType;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.FieldInfo;
+import org.jboss.jandex.IndexView;
+import org.jboss.jandex.MethodInfo;
+
+import io.quarkus.panache.common.Sort;
+
+public class MethodNameParser {
+
+    private static final String ALL_IGNORE_CASE = "AllIgnoreCase";
+    private static final String IGNORE_CASE = "IgnoreCase";
+    private static final String ORDER_BY = "OrderBy";
+
+    private static final List<String> HANDLED_PROPERTY_OPERATIONS = Arrays.asList(
+            "Is", "Equals",
+            "IsNot", "Not",
+            "IsNull", "Null",
+            "IsNotNull", "NotNull",
+            "IsBetween", "Between",
+            "IsLessThan", "LessThan",
+            "IsLessThanEqual", "LessThanEqual",
+            "IsGreaterThan", "GreaterThan",
+            "IsGreaterThanEqual", "GreaterThanEqual",
+            "IsLike", "Like",
+            "IsNotLike", "NotLike",
+            "IsStartingWith", "StartingWith", "StartsWith",
+            "IsEndingWith", "EndingWith", "EndsWith",
+            "IsContaining", "Containing", "Contains",
+            "Before", "IsBefore",
+            "After", "IsAfter",
+            "True", "False",
+            "IsIn", "In",
+            "IsNotIn", "NotIn",
+            "IsEmpty", "Empty",
+            "IsNotEmpty", "NotEmpty");
+
+    private static final Set<String> STRING_LIKE_OPERATIONS = new HashSet<>(Arrays.asList(
+            "IsLike", "Like",
+            "IsNotLike", "NotLike",
+            "IsStartingWith", "StartingWith", "StartsWith",
+            "IsEndingWith", "EndingWith", "EndsWith",
+            "IsContaining", "Containing", "Contains"));
+
+    private static final Set<String> BOOLEAN_OPERATIONS = new HashSet<>(Arrays.asList("True", "False"));
+
+    private final ClassInfo entityClass;
+    private final IndexView indexView;
+
+    public MethodNameParser(ClassInfo entityClass, IndexView indexView) {
+        this.entityClass = entityClass;
+        this.indexView = indexView;
+    }
+
+    public enum QueryType {
+        SELECT,
+        COUNT,
+        EXISTS,
+        DELETE;
+    }
+
+    public Result parse(MethodInfo methodInfo) {
+        String methodName = methodInfo.name();
```

Comment:
It was written from scratch based on Spring Data documentation. 

What do you mean by topic here?

Reply:
I'm referring to this code: https://github.com/quarkusio/quarkus/pull/3489/files#diff-e3c05d6bc601ae34b067ceb2b102311eR93 

Or this one: https://github.com/quarkusio/quarkus/pull/3489/files#diff-e3c05d6bc601ae34b067ceb2b102311eR85

I think you need to match tokens because you could have `Bytes` or `Bylaw` or `Distinctive` for instance.

Using `contains` and `indexOf` is a bit too fragile IMHO.

Reply:
I understand the general concern totally. But I would like to have some concrete method name that would fail.
Thanks

Reply:
I have left this as is for now. I would like to avoid regexs if at all possible. If you have an example where the current implementation would fail, I will happily oblige and switch the implementation

Reply:
After chatting about this, I understood that the current implementation would break if an entity field were named `distinctive` for example. This has now been fixed.
Also a test was added to ensure that an entity field named `bypass` works correctly.